hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/authentication/AuthorizationStatus.java

```diff
@@ -0,0 +1,183 @@
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.fs.azurebfs.authentication;
+
+import org.apache.hadoop.fs.azurebfs.contracts.exceptions.AbfsAuthorizationException;
+import org.apache.hadoop.fs.azurebfs.extensions.AuthorizationResource;
+import org.apache.hadoop.fs.azurebfs.extensions.AuthorizationResourceResult;
+import org.apache.hadoop.fs.azurebfs.extensions.AuthorizationResult;
+
+import java.net.URI;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.HashMap;
+
+import static org.apache.hadoop.fs.azurebfs.constants.FileSystemConfigurations.DEFAULT_SAS_REFRESH_INTERVAL_BEFORE_EXPIRY;
+
+/**
+ * AuthorizationStatus maintains the status of Authorization and also SAS
+ * token if authorizer is providing it.
+ */
+public class AuthorizationStatus {
+  private boolean isAuthorized;
+  private HashMap<URI, SasTokenData> sasTokenMap;
+
+  public AuthorizationStatus() {
+    sasTokenMap = new HashMap<>();
+  }
+
+  /**
+   * Fetch the SAS token
+   *
+   * @return
+   */
+  public String getSasTokenQuery(URI storePathUri) {
+    if (sasTokenMap.containsKey(storePathUri)) {
+      SasTokenData sasTokenData = sasTokenMap.get(storePathUri);
+      if (isValidSas(sasTokenData)) {
+        return sasTokenData.sasToken;
+      }
+    }
+
+    return null;
+  }
+
+  /**
+   * Update authTokenMap
+   * Also update the refresh interval for each SAS token.
+   *
+   * @param authorizationResource
+   * @param authResult - Authorizer AuthorizationResult
+   */
+  public void setSasToken(AuthorizationResource[] authorizationResource,
+      AuthorizationResult authResult) throws AbfsAuthorizationException {
+
+    AuthorizationResourceResult[] resourceResult = authResult
+        .getAuthResourceResult();
+
+    int i = 0;
+    for (AuthorizationResourceResult singleResourceAuth : resourceResult) {
+      // First check if the requested resource matches the resource
+      // for which authToken is returned
+      AuthorizationResource authorizationRequestedForResource =
+          authorizationResource[i];
+      if ((singleResourceAuth == null) || (singleResourceAuth.storePathUri
+          == null) || (singleResourceAuth.authorizerAction == null))  {
+        throw new AbfsAuthorizationException("Invalid authorization response");
```

Comment:
could a resource path go in here? Be useful later

Reply:
Have put in detailed checks.